From 1495246a61cf4ff7e4c648f7ff4b27243adf4aaf Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Tue, 19 Mar 2024 15:51:37 +0100
Subject: [PATCH 0705/1695] plat-stm32mp1: stm32mp_pmic: handle st,mask-reset

Fix test "if (priv->flags & PMIC_REGU_FLAG_MASK_RESET)" that was
always true.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0c2dcdf03ff873b0ee76deac1052f78ebaba95d9 plat-stm32mp1: stm32mp_pmic: handle st,mask-reset]
Change-Id: Ib2a47eeac4dd8c4a2387d677e1911f7c9c02dc98
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/367951
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
index dccc731ba..e8f8b7733 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
@@ -39,13 +39,10 @@
 
 #define PMIC_REGU_COUNT			14
 
-enum {
-	PMIC_REGU_FLAG_MASK_RESET = 0,
-	PMIC_REGU_FLAG_COUNT
-};
+/* PMIC private flags */
+#define PMIC_REGU_FLAG_MASK_RESET	BIT(0)
 
 static_assert(IS_ENABLED(CFG_DRIVERS_REGULATOR));
-static_assert(PMIC_REGU_FLAG_COUNT <= UINT_MAX);
 
 /*
  * struct pmic_regulator_data - Platform specific data
-- 
2.47.2

